appdatabase/migrations/sql/1685134628_transfers_receipt_json_blob_out.up.sql

```diff
@@ -0,0 +1,23 @@
+-- This migration is done in GO code as a custom step.
+-- This file serves as an anchor for the migration system
+-- Check migrateWalletJsonBlobs from appdatabase/database.go
+
+-- The following steps are done in GO code:
+-- ALTER TABLE transfers ADD COLUMN status INT;
+-- ALTER TABLE transfers ADD COLUMN receipt_type INT;
+-- ALTER TABLE transfers ADD COLUMN cumulative_gas_used INT;
+-- ALTER TABLE transfers ADD COLUMN contract_address TEXT;
+-- ALTER TABLE transfers ADD COLUMN gas_used INT;
+-- ALTER TABLE transfers ADD COLUMN tx_index INT;
+-- ALTER TABLE transfers ADD COLUMN protected BOOLEAN;
+-- ALTER TABLE transfers ADD COLUMN gas_limit UNSIGNED INT;
+-- ALTER TABLE transfers ADD COLUMN gas_price_clamped64 INT;
+-- ALTER TABLE transfers ADD COLUMN gas_tip_cap_clamped64 INT;
+-- ALTER TABLE transfers ADD COLUMN gas_fee_cap_clamped64 INT;
+-- ALTER TABLE transfers ADD COLUMN amount_padded128hex CHAR(32);
+-- ALTER TABLE transfers ADD COLUMN account_nonce INT;
+-- ALTER TABLE transfers ADD COLUMN size INT;
+
+-- CREATE INDEX transfers_status ON transfers (status);
+
```

Comment:
I tried to explain it in the comments here: https://github.com/status-im/status-go/blob/d8b24ca160bd52df99d49dc09a903d3cd6914590/sqlite/migrate.go#L26

The migration API doesn't support rolling back to the previous migration to leave the DB in a consistent state (in case of custom golang migration failing). Therefore, I use migration down to roll back to the previous migration (before the custom one). The `down` migration for the current custom migration can be done in SQL. Still, it is more complicated than a simple `tx.Rollback` in the go code (have to remove rows, create a temporary DB to drop the new columns and recopy data). However, for another type of migration, providing an `up` SQL file and a `down` SQL file that discards changes made by the `up` one, might be a better idea and it will work the same.

Hope this clarifies your concerns.